app/forms/state_file/nj_eligibility_health_insurance_form.rb

```diff
@@ -1,11 +1,22 @@
 module StateFile
   class NjEligibilityHealthInsuranceForm < QuestionsForm
     set_attributes_for :intake, :eligibility_all_members_health_insurance
+    attr_accessor :dependents_attributes
+    delegate :dependents, to: :intake
 
     validates :eligibility_all_members_health_insurance, presence: true
 
     def save
       @intake.update(attributes_for(:intake))
+      if @intake.eligibility_all_members_health_insurance_no?
+        @intake.update!(dependents_attributes: dependents_attributes.to_h)
+      else
+        attributes_array = intake.dependents.map do |dependent|
+           { id: dependent.id, nj_did_not_have_health_insurance: 'no' }
+        end
+        attributes_hash = Hash[(0...attributes_array.size).zip attributes_array]
```

Comment:
[sand?] :) this is slick, and I appreciate that you've simplified the approach you saw elsewhere in our codebase! i just spent a little while poking at it, and i realized that update can actually just take an array of nested attributes (see docs [here](https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html#:~:text=the%20updated%20hash.-,One%2Dto%2Dmany,-Consider%20a%20member))! i'm guessing that the support for this hash format that you're recreating (and that you've seen us working with elsewhere) is just to work around the fact that it is the format sent in the params hash by rails forms.

so, i think you can just remove `attributes_hash` entirely here and call `@intake.update!(dependents_attributes: attributes_array)` on the next line. i tried it in your branch and the tests still pass, though please double check me to make sure i'm not missing something.